test_communication/test/test_invalid_secure_node_creation_c.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright 2016-2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <limits>
+#include <memory>
+#include <string>
+
+#include "rcl/rcl.h"
+#include "rcl/error_handling.h"
+#include "rcutils/get_env.h"
+
+#ifndef SCOPE_EXIT_HPP_
+#define SCOPE_EXIT_HPP_
+
+template<typename Callable>
+struct ScopeExit
+{
+  explicit ScopeExit(Callable callable)
+  : callable_(callable) {}
+  ~ScopeExit() {callable_(); }
+
+private:
+  Callable callable_;
+};
+
+template<typename Callable>
+ScopeExit<Callable>
+make_scope_exit(Callable callable)
+{
+  return ScopeExit<Callable>(callable);
+}
+
+void custom_putenv(const char * name, const char * value)
+{
+#ifdef WIN32
+  _putenv_s(name, value);
+#else
+  setenv(name, value, 1);
+#endif
```

Comment:
this will be moved to rcultils in a later PR, I didn't have time to test it properly / add testing for it so I kept it in here for now